src/System.Text.Json/src/System/Text/Json/JsonElement.cs

```diff
@@ -0,0 +1,458 @@
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        [CLSCompliant(false)]
+        public ulong GetUInt64()
+        {
+            if (TryGetUInt64(out ulong value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public bool TryGetDouble(out double value)
+        {
+            CheckValidInstance();
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public double GetDouble()
+        {
+            if (TryGetDouble(out double value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public bool TryGetSingle(out float value)
+        {
+            CheckValidInstance();
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public float GetSingle()
+        {
+            if (TryGetSingle(out float value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public bool TryGetDecimal(out decimal value)
+        {
+            CheckValidInstance();
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public decimal GetDecimal()
+        {
+            if (TryGetDecimal(out decimal value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        internal string GetPropertyName()
+        {
+            CheckValidInstance();
+
+            return _parent.GetNameOfPropertyValue(_idx);
+        }
+
+        public string GetRawText()
+        {
+            CheckValidInstance();
+
+            return _parent.GetRawValueAsString(_idx);
+        }
+
+        internal string GetPropertyRawText()
+        {
+            CheckValidInstance();
+
+            return _parent.GetPropertyRawValueAsString(_idx);
+        }
+
+        public ArrayEnumerator EnumerateArray()
+        {
+            CheckValidInstance();
+
+            JsonTokenType tokenType = TokenType;
+
+            if (tokenType != JsonTokenType.StartArray)
+            {
+                throw ThrowHelper.GetJsonElementWrongTypeException(JsonTokenType.StartArray, tokenType);
+            }
+
+            return new ArrayEnumerator(this);
+        }
+
+        public ObjectEnumerator EnumerateObject()
+        {
+            CheckValidInstance();
+
+            JsonTokenType tokenType = TokenType;
+
+            if (tokenType != JsonTokenType.StartObject)
+            {
+                throw ThrowHelper.GetJsonElementWrongTypeException(JsonTokenType.StartObject, tokenType);
+            }
+
+            return new ObjectEnumerator(this);
+        }
+
+        public override string ToString()
+        {
+            switch (TokenType)
+            {
+                case JsonTokenType.None:
+                case JsonTokenType.Null:
+                    return string.Empty;
```

Comment:
Should we output "null" for `JsonTokenType.Null`?

Reply:
string.Empty was chosen to match JSON.Net.

Reply:
I don't know why Json.NET behaves this way. I think explicitly outputting null makes more sense, tbh, just like the true/false literals.

cc @JamesNK